### PR TITLE
feat:[SSCA-1058]: Added access control for remediation tracker

### PR DESCRIPTION
**Pull request by humanshu2002**
## Describe your changes
Add access control client and config in SSCA, and access for Remediation Tracker.

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/56676)

### CODE REVIEW

**Comment by humanshu2002** [comment]
trigger uts

**Comment by humanshu2002** [comment]
trigger uts

**Comment by humanshu2002** [comment]
trigger uts

**Review by pranay-harness — DISMISSED** [review]

**Comment by humanshu2002** [comment]
trigger uts

File: access-control/service/src/main/resources/io/harness/accesscontrol/roles/managed-roles.yml [hunk 1]
```diff
@@ -1250,4 +1250,15 @@ roles:
         - idp_advancedconfiguration_delete
       managed: true
       allowedScopeLevels:
-        - account
\ No newline at end of file
+        - account
+    - identifier: _ssca_remediation_tracker
+      name: SSCA Remediation Tracker
```
Comment (reetika04) [inline access-control/service/src/main/resources/io/harness/accesscontrol/roles/managed-roles.yml:1255]: The role name isn't supposed to be SSCA Remediation Tracker. We can add this as `SSCA Remediation Admin` or `SSCA Remediation Manager` Please check with @KapilDigani

File: access-control/service/src/main/resources/io/harness/accesscontrol/roles/managed-roles.yml [hunk 2]
```diff
@@ -1250,4 +1250,15 @@ roles:
...
+      description: Can create, view, update and close SSCA Remediation Tracker
+      permissions:
+        - ssca_remediation_tracker_create
+        - ssca_remediation_tracker_view
+        - ssca_remediation_tracker_update
+        - ssca_remediation_tracker_close
```
Comment (reetika04) [inline access-control/service/src/main/resources/io/harness/accesscontrol/roles/managed-roles.yml:1261]: Why do we need both create and update, only have update single permission. We can have a single manage permission as well.

File: access-control/service/src/main/resources/io/harness/accesscontrol/roles/managed-roles.yml [hunk 3]
```diff
@@ -1250,4 +1250,15 @@ roles:
         - idp_advancedconfiguration_delete
       managed: true
       allowedScopeLevels:
-        - account
\ No newline at end of file
+        - account
+    - identifier: _ssca_remediation_tracker
```
Comment (reetika04) [inline access-control/service/src/main/resources/io/harness/accesscontrol/roles/managed-roles.yml:1254]: Update the version of this file.

File: access-control/service/src/main/resources/io/harness/accesscontrol/permissions/permissions.yml [hunk 4]
```diff
@@ -1903,4 +1903,36 @@ permissions:
     allowedScopeLevels:
       - account
       - organization
-      - project
\ No newline at end of file
+      - project
+  - identifier: ssca_remediation_tracker_create
```
Comment (reetika04) [inline access-control/service/src/main/resources/io/harness/accesscontrol/permissions/permissions.yml:1907]: Update the version for the file

File: ssca-manager/chart/templates/config.yaml [hunk 5]
```diff
@@ -43,6 +43,8 @@ data:
   PIPELINE_SERVICE_CLIENT_BASEURL: http://pipeline-service.{{ .Release.Namespace }}.svc.cluster.local:12001/api/
   MANAGER_CLIENT_BASEURL: http://harness-manager.{{ .Release.Namespace }}.svc.cluster.local:9090/api/
   POLICY_MGMT_SERVICE_BASEURL: http://policy-mgmt.{{ .Release.Namespace }}.svc.cluster.local:3000/
+  ACCESS_CONTROL_ENABLED: true
+  ACCESS_CONTROL_BASE_URL: 'http://access-control.{{ .Release.Namespace }}.svc.cluster.local:9006/api/'
   {{ include "harnesscommon.harnessfunctions.java17flags" (dict "context" $) | nindent 2 }}
```
Comment (reetika04) [inline ssca-manager/chart/templates/config.yaml:48]: Update the chart version as well.

**Review by reetika04 — CHANGES_REQUESTED** [review]

**Comment by humanshu2002** [comment]
trigger uts

File: access-control/service/src/main/resources/io/harness/accesscontrol/roles/managed-roles.yml [hunk 6]
```diff
@@ -1250,4 +1250,4 @@ roles:
         - idp_advancedconfiguration_delete
       managed: true
       allowedScopeLevels:
-        - account
\ No newline at end of file
+        - account
```
Comment (reetika04) [inline access-control/service/src/main/resources/io/harness/accesscontrol/roles/managed-roles.yml:1253]: please remove changes in this file

Comment (humanshu2002) [reply access-control/service/src/main/resources/io/harness/accesscontrol/roles/managed-roles.yml:1253]: it was this only earlier

**Comment by humanshu2002** [comment]
trigger uts

**Comment by humanshu2002** [comment]
trigger ut6

**Comment by humanshu2002** [comment]
trigger uts

**Comment by humanshu2002** [comment]
trigger ut5,ut6,ut7,ut8,ut9,ut4

File: dockerization/manager/scripts/replace_configs.sh [hunk 7]
```diff
@@ -1271,3 +1271,6 @@ replace_key_value awsServiceEndpointUrls.cloudwatchEndPointUrl "$AWS_SERVICE_END
 
 replace_key_value maxAccountsToDeleteInParallel "$MAX_ACCOUNTS_TO_DELETE_IN_PARALLEL"
 
+replace_key_value accessControlClientConfig.enableAccessControl "$ACCESS_CONTROL_ENABLED"
```
Comment (reetika04) [inline dockerization/manager/scripts/replace_configs.sh:1274]: This is not where the replace_configs go @humanshu2002 . Please update this in ssca-manager/build/container/scripts/replace_configs.sh

**Review by reetika04 — CHANGES_REQUESTED** [review]

**Review by reetika04 — APPROVED** [review]

**Comment by humanshu2002** [comment]
trigger uts

**Review by arya-harness — APPROVED** [review]

**Review by jyotiraval — APPROVED** [review]

**Comment by humanshu2002** [comment]
trigger ut8,ut0

**Comment by humanshu2002** [comment]
trigger ut0

**Comment by humanshu2002** [comment]
triggger ut8

**Comment by humanshu2002** [comment]
trigger ut8

**Comment by humanshu2002** [comment]
trigger ut8

**Pull request merged**